shell/components/SortableTable/index.vue

```diff
@@ -318,6 +353,20 @@ export default {
 
     this._onScroll = this.onScroll.bind(this);
     $main.on('scroll', this._onScroll);
+
+    this.updateLiveAndDelayed();
+
+    // check if user clicked outside the advanced filter box
+    document.addEventListener('click', this.onClickOutside);
+
+    // register watcher to watch rows for columnOptions
+    if (this.hasAdvancedFiltering) {
+      this.columnOptions = this.setColsOptions();
+
+      this.colOptionsWatcher = this.$watch('rows', function() {
+        this.columnOptions = this.setColsOptions();
```

Comment:
You are right with your assessment. It requires prudence tackling such a watcher, especially on large datasets.

This was never tested at scale, tbh.

Unfortunately a watcher is needed because we need to set/change the cols options based on the labels for which we need to iterate over all the rows in order to find out which labels exist. I you have any ideas to circumvent this watcher I am all for it.

we have 2 good points in our favour right now:
- the watcher is only added if we have advanced filtering
- the watcher doesn't trigger table re-renders, which is very costly, especially on those delayed cols. It only updates the DOM part regarding the filter selection and col visibility